dbt/adapters/bigquery/connections.py

```diff
@@ -292,8 +298,29 @@ def exception_handler(self, sql):
                 exc_message = exc_message.split(BQ_QUERY_JOB_SPLIT)[0].strip()
             raise DbtRuntimeError(exc_message)
 
-    def cancel_open(self) -> None:
-        pass
+    def cancel_open(self) -> List[str]:
+        names = []
+        this_connection = self.get_if_exists()
+        with self.lock:
+            for thread_id, connection in self.thread_connections.items():
+                if connection is this_connection:
+                    continue
+
+                if connection.handle is not None and connection.state == ConnectionState.OPEN:
+                    client = connection.handle
+                    for job_id in self.jobs_by_thread.get(thread_id, []):
+
+                        def fn():
+                            return client.cancel_job(job_id)
+
+                        self._retry_and_handle(msg=f"Cancel job: {job_id}", conn=connection, fn=fn)
+
+                    self.close(connection)
+
+                if connection.name is not None:
```

Comment:
Do we specifically want all connections which are not `this_connection`? Or do we only want connections in which we cancelled jobs? In this current flow, a connection for which `connection.state == ConnectionState.CLOSED` will show up in `names`, which doesn't feel like an intuitive list to get from `cancel_open'.

Reply:
If i backtrack `open_cancel` to core it looks like one of the only places we call it is for `cancel_open_connections` which does make me think the desired result is that all closed connections are accounted for?

Reply:
In other words we should only be returning the connections which we cancelled during this call, right?